pythainlp/corpus/common.py

```diff
@@ -336,3 +337,26 @@ def thai_synonyms() -> dict:
 def thai_synonym() -> dict:
     warnings.warn("Deprecated: Use thai_synonyms() instead.", DeprecationWarning)
     return thai_synonyms()
+
+
+def find_synonyms(word) -> Union[List[str], None]:
```

Comment:
To make the type checking more straightforward, we can just return an empty list.

I would suggest:
`def find_synonyms(word: str) -> List[str]:`

and in case of no synonyms, return an empty list:

`return []`